DEVELOPER_GUIDE.md

```diff
@@ -422,6 +423,88 @@ Output plugins are defined in [flb_output.h](https://github.com/fluent/fluent-bi
 
 The [stdout plugin](plugins/out_stdout) is very simple; review its code to understand how output plugins work.
 
+#### Config Maps
+
+Config maps are an improvement to the previous Fluent Bit API that was used by plugins to read configuration values. The new config maps feature warns the user if there is an unknown configuration key and reduces risk of bad configuration due to typos or deprecated property names. They will also allow dynamic configuration reloading to be implemented in the future.
+
+There are various types of supported configuration types. Full list available [here](https://github.com/fluent/fluent-bit/blob/master/include/fluent-bit/flb_config_map.h#L29). The most used ones are :
+
+| Type                   | Description           | 
+| -----------------------|:---------------------:| 
+| FLB_CONFIG_MAP_INT     | Represents integer data type | 
+| FLB_CONFIG_MAP_BOOL    | Represents boolean data type | 
+| FLB_CONFIG_MAP_DOUBLE  | Represents a double |
+| FLB_CONFIG_MAP_SIZE    | Provides size_type as an integer datatype large enough to represent any possible string size. |
+| FLB_CONFIG_MAP_STR     | Represents string data type |
+| FLB_CONFIG_MAP_CLIST   | Comma separated list of strings |
+| FLB_CONFIG_MAP_SLIST   | Empty space separated list of strings |
+
+A config map expects certain public fields at registration. They are 
+
+| Public Fields | Description           | 
+| --------------|:---------------------| 
+| Type          | This field requires the data type of the property that we are writing to the config map. If the property is of type `int` we use `FLB_CONFIG_MAP_INT`, if `string` `FLB_CONFIG_MAP_STR` etc. |
+| Name          | This field requires the name of the configuration property. For example for the property flush count we use `flush_count`|
+| Default Value | This field allows the user to set the default value of the property. For example, for a property of type `FLB_CONFIG_MAP_BOOL` (boolean), the default value may be false. Then we have to give `false` as default value. If there is no default value, `NULL` is given.|
```

Comment:
>  Then we have to give `false` as default value

FLB_FALSE?

Reply:
Yes but in config map we send "false" or "true"
Refer these :
https://github.com/fluent/fluent-bit/blob/master/src/flb_config_map.c#L150
https://github.com/fluent/fluent-bit/blob/master/src/flb_utils.c#L459
Example -
https://github.com/fluent/fluent-bit/blob/master/plugins/out_forward/forward.c#L1153